main.cpp

```diff
@@ -58,49 +56,68 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* n);   //1
+    int value;
+    std::string name;
 };
 
-struct <#structName1#>                                //4
+T::T(int v, const char* n) : value(v), name(n) {}
+
+struct Compare                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if(a != nullptr)
+        {
+            if( a->value < b->value ) return a;
+        }
+        if(b != nullptr)
+        {
+            if( a->value > b->value ) return b;
+        }
```

Comment:
You're using `b` without checking if `b` is a nullptr in the 1st `if` statement.
and same goes for `a` in the 2nd `if` statement.
you should check both variables before you do either `if` statement: 
```
if( ... )
{
    if( a < b ) ...
    if( b < a ) ...
}
```
- [ ] revise